Unable to tokenize hex string with Rust tokenizer using Snowflake dialect

Follow-up from #4559, filing the reproducible issue here.

**Fully reproducible code snippet**

Note: this issue seems to be related to the **Rust tokenizer**.

The following piece of code (tested against Snowflake) ...:
```
from sqlglot import parse_one

query = "INSERT INTO test VALUES (X'48FAF43B0AFCEF9B63EE3A93EE2AC2')"
parse_one(query, dialect="snowflake")
```

... fails with:
```
E           sqlglot.errors.TokenError: Error tokenizing 'NTO test VALUES (X'48FAF43B0AFCEF9B63EE3A93EE2AC2'': Numeric string contains invalid characters from 1:25
```

--- a/sqlglotrs/src/tokenizer.rs
+++ b/sqlglotrs/src/tokenizer.rs
@@ -458,7 +458,7 @@ impl<'a> TokenizerState<'a> {
         let text = self.extract_string(&end, false, token_type == self.token_types.raw_string, true)?;
 
         if let Some(b) = base {
-            if u64::from_str_radix(&text, b).is_err() {
+            if u128::from_str_radix(&text, b).is_err() {
                 return self.error_result(format!(
                     "Numeric string contains invalid characters from {}:{}",
                     self.line, self.start

--- a/tests/dialects/test_snowflake.py
+++ b/tests/dialects/test_snowflake.py
@@ -21,6 +21,7 @@ class TestSnowflake(Validator):
         expr.selects[0].assert_is(exp.AggFunc)
         self.assertEqual(expr.sql(dialect="snowflake"), "SELECT APPROX_TOP_K(C4, 3, 5) FROM t")
 
+        self.validate_identity("INSERT INTO test VALUES (x'48FAF43B0AFCEF9B63EE3A93EE2AC2')")
         self.validate_identity("exclude := [foo]")
         self.validate_identity("SELECT CAST([1, 2, 3] AS VECTOR(FLOAT, 3))")
         self.validate_identity("SELECT CONNECT_BY_ROOT test AS test_column_alias")

